controllers/rcj_soccer_referee_supervisor/referee/event_handlers.py

```diff
@@ -0,0 +1,109 @@
+import json
+from pathlib import Path
+from datetime import datetime
+from typing import Optional
+
+from referee.consts import GameEvents
+
+
+class EventHandler:
+    def __init__(self):
+        pass
+
+    def handle(
+        self,
+        supervisor,  # Supervisor from supervisor.py
+        type: str,
+        payload: Optional[dict] = None,
+    ):
+        """Handle the incoming event
+
+        Args:
+            supervisor (Supervisor): Referee supervisor
+            type (str): Event type
+            payload (dict, optional): More information about the event
+        """
+        raise NotImplementedError
+
+
+class JSONLoggerHandler(EventHandler):
+    """Handler for writing data to json file."""
+
+    def __init__(self, logfile: Path):
+        super().__init__()
+        self.logfile = logfile
+
+    def handle(self, supervisor, type: str, payload: Optional[dict] = None):
+        matchtime = supervisor.match_time - supervisor.time
+        data = {
+            "datetime": datetime.utcnow().strftime('%Y-%m-%dT%H:%M:%S.%fZ'),
+            "matchtime": matchtime,
+            "event": type,
+        }
+
+        if payload is not None:
+            data['payload'] = payload
+
+        with self.logfile.open('a') as outfile:
+            json.dump(data, outfile)
+            outfile.write('\n')
+
+
+class DrawMessageHandler(EventHandler):
+    """Handler for creating the message which is drawn onto world window."""
+
+    def create_inside_penalty_for_too_long_msg(
+        self,
+        robot_name: str,
+        **kwargs,
+    ) -> str:
+        return f"Robot {robot_name}: Inside penalty for too long."
+
+    def create_lack_of_progress_msg(
+        self,
+        type: str,
+        robot_name: Optional[str] = None,
+        **kwargs,
+    ) -> str:
+        if type == "ball":
+            return "Ball: Lack of progress."
+        return f"Robot {robot_name}: Lack of progress."
+
+    def create_goal_msg(self, team_name: str, **kwargs) -> str:
+        return f"A goal was scored by {team_name}."
+
+    def create_kickoff_msg(self, robot_name: str, **kwargs) -> str:
+        return f"Robot {robot_name} is kicking off."
+
+    def create_match_start_msg(self, total_match_time: int, **kwargs) -> str:
+        return f"The match ({total_match_time}s) has started."
+
+    def create_match_finish_msg(self, total_match_time: int, **kwargs) -> str:
+        return f"The match time {total_match_time}s is over."
+
+    def handle(self, supervisor, type: str, payload: Optional[dict] = None):
+        event_mapping = {
```

Comment:
We somehow need to map event to method which creates the resulting message.
The other way could be 

```python
def create_event_type_msg():
    ...

method = getattr(self, f"create_{EVENT_TYPE.lower()}_msg")
```

Would that be

Reply:
@Adman Right, I see. If all we do is this mapping (i.e. there are no additional arguments to be passed), I guess it may be a bit more straightforward.

I don't have a very well formed opinion here though.

Reply:
Done.